commands.go

```diff
@@ -110,7 +110,16 @@ var Commands = []cli.Command{
 
 			host, err := store.Create(name, driver, c)
 			if err != nil {
-				log.Fatal(err)
+				log.Errorf("Error creating host: %s", err)
+				if c.GlobalBool("debug") {
+					log.Fatal(err)
+				} else {
+					log.Warnf("Removing provisioned machine.  Use --debug to prevent removal.")
+					// we know there was an error so do not check for the error on removal
+					// instead we will Fatal with a message to prevent spamming with error messages
+					store.Remove(name, true)
```

Comment:
Interesting - I wonder why that is.  At any rate for this PR we can probably just use `_` (to visually demarcate that we're conciously discarding the error) and then later rework `store.Create` to be a little more flexible (since I don't think an error should be returned in this case, although there may well be a good reason I'm overlooking).